src/main/java/org/jenkinsci/plugins/gitclient/CleanCommand.java

```diff
@@ -0,0 +1,26 @@
+package org.jenkinsci.plugins.gitclient;
+
+import java.util.List;
+
+/**
+ * Command to clean a repository.
+ *
+ */
+public interface CleanCommand extends GitCommand {
+    /**
+     * exclude patterns.
+     *
+     * @param excludePatterns a {@link java.util.List} with exclude patterns.
+     * @return a {@link org.jenkinsci.plugins.gitclient.CleanCommand} object.
+     */
+    CleanCommand excludePatterns(List<String> excludePatterns);
+
```

Comment:
Could you add a method submodules(boolean removeSubmodules) to allow callers to specify that directories which contain a `.git` directory should also be cleaned?